x-pack/plugins/observability/server/lib/annotations/bootstrap_annotations.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as t from 'io-ts';
+import { CoreSetup, PluginInitializerContext, KibanaRequest, RequestHandler } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { PathReporter } from 'io-ts/lib/PathReporter';
+import { isLeft } from 'fp-ts/lib/Either';
+import {
+  getAnnotationByIdRt,
+  createAnnotationRt,
+  deleteAnnotationRt,
+} from '../../../common/annotations';
+import { PromiseReturnType } from '../../../../apm/typings/common';
+import { createAnnotationsClient } from './create_annotations_client';
+
+interface Params {
+  index: string;
+  core: CoreSetup;
+  context: PluginInitializerContext;
+}
+
+export type ScopedAnnotationsClientFactory = PromiseReturnType<
+  typeof bootstrapAnnotations
+>['getScopedAnnotationsClient'];
+
+export type ScopedAnnotationsClient = ReturnType<ScopedAnnotationsClientFactory>;
+export type AnnotationsAPI = PromiseReturnType<typeof bootstrapAnnotations>;
+
+const unknowns = schema.object({}, { unknowns: 'allow' });
+
+export async function bootstrapAnnotations({ index, core, context }: Params) {
+  const logger = context.logger.get('annotations');
+
+  function wrapRouteHandler<TType extends t.Type<any>>(
```

Comment:
This needs some work I think. Open to suggestions to make it better.

Reply:
Perhaps extract everything related to the API into a create_anotations_api.ts so it's clear that bootstrap_annotations creates the API and a returns a client as well.

Reply:
moved it into `register_annotation_apis` (named it as such to distinguish it from `create_annotations_client`)